src/main/java/sign_up/use_case/SignUpInteractor.java

```diff
@@ -0,0 +1,80 @@
+package sign_up.use_case;
+
+import entities.CurrentUser;
+import entities.User;
+import sign_up.use_case.exceptions.MakeUserException;
+
+public class SignUpInteractor implements SignUpInputBoundary{
+    // An object that accesses the database
+    private final SignUpDataAccessInterface dataAccess;
+
+    // An object that accesses the UI
+    private final SignUpOutputBoundary outputBoundary;
+
+    // An object whose job is to create User elements
+    private final UserFactory userFactory;
+
+    public SignUpInteractor(SignUpDataAccessInterface dataAccess, SignUpOutputBoundary outputBoundary){
+        this.dataAccess = dataAccess;
+        this.outputBoundary = outputBoundary;
+        this.userFactory = new UserFactory();
+    }
+
+    @Override
+    public void signUp(SignUpRequestModel requestModel) {
+        try {
+            SignUpResponseModel responseModel = this.signUpHelper(requestModel);
+            this.outputBoundary.present(responseModel);
+        }
+        catch (MakeUserException e) {
+            SignUpResponseModel responseModel = new SignUpResponseModel(false, e.getMessage());
+        }
+
+
+    }
+
+    private SignUpResponseModel signUpHelper(SignUpRequestModel requestModel) throws MakeUserException {
+        if (requestModel.email.isEmpty() || requestModel.password.isEmpty()) {
```

Comment:
Shouldn't be accessing fields of another class directly imo. Make these attributes private in the corresponding class and add getters and call them here (and elsewhere where the same idea applies).